### PR TITLE
feat(automation): add audit reviewer for batch PR analysis

### LINKED ISSUE #994: feat(automation): add audit reviewer for batch PR analysis with documentation and integration tests
## Summary

Add comprehensive documentation and integration tests for the audit reviewer (`hephaestus-audit-prs`), a coordinator+sub-agent dispatch pattern that reviews ALL open PRs in one invocation.

### Documentation
- New `docs/audit-reviewer.md` covering architecture, CLI usage, programmatic API, report format, coordinator prompt, error handling, env vars, and examples
- Updated `docs/index.md` to link to the new audit reviewer page

### Integration tests (11 new)
- Module importability and key symbol exports
- `_parse_coordinator_results` with realistic output + multi-block JSON
- `write_audit_report` with valid JSON + posting failures
- `print_audit_summary` with per-PR log line verification
- `AuditReviewer` construction (default, codex, pr_numbers, state_dir)

### Unit tests (expanded from 24 to 43)
- Edge cases for `_parse_coordinator_results` (empty, whitespace, prose, extra fields, malformed fence)
- Tests for `run_audit_coordinator` (Claude/Codex paths, dry_run, error handling)
- Tests for `AuditReviewer.run()` (happy path, no PRs, coordinator failure, posting failure, pr_numbers, dry run)
- Test for `_fetch_prs_by_number` with empty input

### Orchestration smoke test
- Added audit_reviewer to OMITTED_MODULES and hephaestus-audit-prs to CONSOLE_SCRIPTS in test_orchestration_smoke.py

## Definition of Done
- [x] 54 audit reviewer tests pass (43 unit + 11 integration)
- [x] Lints clean (ruff check)
- [x] Documentation page exists and is linked from index
- [x] All tests pass and lints are clean

### PR DESCRIPTION
**Pull request by mvillmow**
## Summary

Add comprehensive audit reviewer implementation for batch PR analysis with 54 tests (43 unit + 11 integration) and full documentation.

- **New module**: `hephaestus/automation/audit_reviewer.py` — coordinator-pattern auditor that reviews ALL open PRs in one invocation
- **New public helper**: `fetch_open_prs()` in github_api.py with `--limit 0` (no row cap)
- **54 tests**: Unit tests cover JSON parsing, report writing, coordinator invocation, CLI parsing, and error handling; integration tests verify module structure, real disk I/O, and agent output parsing
- **Documentation**: New `docs/audit-reviewer.md` page with architecture, CLI usage, programmatic API, report format, and error handling
- **Console script**: `hephaestus-audit-prs` registered in pyproject.toml
- **Lazy exports**: `AuditReviewer` added to hephaestus.automation exports
- **Smoke test updates**: Added audit_reviewer module and console script to orchestration test

## Testing

- **Unit tests (43)**: `_parse_coordinator_results` (11), `write_audit_report` (4), `print_audit_summary` (3), `_build_coordinator_prompt` (3), `_fetch_prs_by_number` (3), `run_audit_coordinator` (7), `AuditReviewer.run()` (7), CLI parser (5)
- **Integration tests (11)**: Module importability, class exports, realistic multi-PR parsing, disk I/O, posting failure isolation, log output, and constructor variants
- **All tests pass**: 54/54 audit reviewer tests + full suite (3573 tests)
- **Linting**: ruff check and ruff format both clean

Closes #994

### CODE REVIEW

File: hephaestus/automation/github_api.py [hunk 1]
```diff
@@ -1240,6 +1240,28 @@ def fetch_issue_info(issue_number: int) -> IssueInfo:
...
+    result = _gh_call(
+        [
+            "pr",
+            "list",
+            "--state",
+            "open",
+            "--json",
+            "number,title,headRefName,url,isDraft",
+            "--limit",
+            "0",
```
Comment (mvillmow) [inline hephaestus/automation/github_api.py]: `--limit 0` is not gh's no-cap sentinel — it is a literal row limit of zero, so `fetch_open_prs()` will always return an empty list. The gh CLI does not document 0 as unlimited; use `--limit 2147483647` (INT_MAX) or remove the `--limit` flag entirely and rely on pagination (add `--paginate` if needed). Reproduce: `gh pr list --state open --limit 0` returns `[]` on any repo with open PRs.

**Review by mvillmow — COMMENTED** [review]
POLICY VIOLATION: Closes (PR description empty, no Closes #994 line), auto-merge (enabled=true must be false), signed-commits (commit signing state array is empty). Additionally: fetch_open_prs() uses --limit 0 which returns zero rows on gh CLI; silent-empty guard logic deviates from spec.

File: hephaestus/automation/audit_reviewer.py [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+        log_file.write_text(response)
+    except subprocess.CalledProcessError as e:
+        log_file.write_text(f"EXIT {e.returncode}\nSTDOUT:\n{e.stdout}\nSTDERR:\n{e.stderr}")
+        raise RuntimeError(f"Audit coordinator failed: {e.stderr or e.stdout}") from e
+    except subprocess.TimeoutExpired as e:
+        log_file.write_text(f"TIMEOUT after {e.timeout}s\n{e.output or ''}")
+        raise RuntimeError("Audit coordinator timed out") from e
+
+    audits = _parse_coordinator_results(response)
+    if response.strip() and _FENCE_RE.search(response) is None:
```
Comment (mvillmow) [inline hephaestus/automation/audit_reviewer.py]: The guard condition diverges from the plan spec and is weaker than intended. The plan says: raise when `response.strip()` is non-empty but `audits` (the parsed result) is empty. The code raises only when no fence is present at all (`_FENCE_RE.search(response) is None`). A response with a fence containing an unrecognized structure — e.g. `{"not_audits": 42}` (no `audits` key, no `pr_number` key) — silently returns `[]` instead of raising. Fix: `if response.strip() and not audits: raise RuntimeError("Coordinator returned no parseable JSON block")`

File: hephaestus/automation/audit_reviewer.py [hunk 3]
```diff
@@ -0,0 +1,267 @@
...
+
+
+def _build_parser() -> argparse.ArgumentParser:
+    parser = argparse.ArgumentParser(
+        prog="hephaestus-audit-prs",
+        description="Audit ALL open PRs in one coordinator agent invocation.",
+    )
+    parser.add_argument(
+        "--pr-numbers",
+        nargs="*",
```
Comment (mvillmow) [inline hephaestus/automation/audit_reviewer.py]: `nargs="*"` for `--pr-numbers` creates a silent footgun: `--pr-numbers` passed without any following integers yields `[]`, which (via the `if self.pr_numbers` branch in `AuditReviewer.run()`) falls through to `fetch_open_prs()` and audits ALL open PRs. A user who types `hephaestus-audit-prs --pr-numbers --dry-run` (forgetting the actual numbers) silently audits every open PR. Use `nargs="+"` to require at least one number when the flag is provided, or add a post-parse validation that rejects an empty `--pr-numbers`.

File: hephaestus/automation/audit_reviewer.py [hunk 4]
```diff
@@ -0,0 +1,267 @@
...
+    """
+    if dry_run:
+        logger.info("[DRY RUN] Would audit %d PR(s)", len(prs))
+        return [
+            {"pr_number": p["number"], "verdict": "UNSURE", "summary": "[DRY RUN]", "findings": []}
+            for p in prs
+        ]
+    if not prs:
+        return []
+    prompt = _build_coordinator_prompt(prs)
```
Comment (mvillmow) [inline hephaestus/automation/audit_reviewer.py:136]: `invoke_claude_with_session(issue=0, ...)` passes `0` as the issue number. The plan flagged this as unverified. If `invoke_claude_with_session` constructs session names, paths, or API calls using the issue number, `0` may produce silent bugs (e.g., a path `build/.issue_implementer/0/` or a malformed session key). Verify that `0` is an accepted sentinel — check `pr_reviewer.py:225` for the call pattern used there and confirm `issue=0` is safe or substitute a sentinel constant.

File: hephaestus/automation/audit_reviewer.py [hunk 5]
```diff
@@ -0,0 +1,267 @@
...
+        log_file.write_text(f"EXIT {e.returncode}\nSTDOUT:\n{e.stdout}\nSTDERR:\n{e.stderr}")
+        raise RuntimeError(f"Audit coordinator failed: {e.stderr or e.stdout}") from e
+    except subprocess.TimeoutExpired as e:
+        log_file.write_text(f"TIMEOUT after {e.timeout}s\n{e.output or ''}")
+        raise RuntimeError("Audit coordinator timed out") from e
+
+    audits = _parse_coordinator_results(response)
+    if response.strip() and not audits:
+        raise RuntimeError("Coordinator returned no parseable JSON block")
+    return audits
```
Comment (mvillmow) [inline hephaestus/automation/audit_reviewer.py:180]: `write_audit_report` is called unconditionally even when `dry_run=True`. The docs state dry-run skips 'the agent call and the GitHub posting step' — writing a report file is a third side effect not mentioned. Either skip `write_audit_report` during dry-run, or document explicitly that the report is always written regardless of `--dry-run`. P7/POLA: the current behavior contradicts the documented dry-run semantics.

**Review by mvillmow — COMMENTED** [review]
POLICY VIOLATION: Closes (PR description empty, no Closes #994), auto-merge (enabled=true before review GO), signed-commits (commit signing state array is empty).

File: tests/unit/automation/test_audit_reviewer.py [hunk 6]
```diff
@@ -0,0 +1,449 @@
...
+    ) -> None:
+        mock_fetch.return_value = [{"number": 100, "title": "Test"}]
+        mock_coord.return_value = [{"pr_number": 100, "verdict": "GO", "summary": "Good"}]
+        reviewer = AuditReviewer(dry_run=True)
+        _, _ = reviewer.run()
+        assert mock_coord.call_args[1]["dry_run"] is True
+
+    def test_state_dir_default_under_build(self, tmp_path: Path) -> None:
+        reviewer = AuditReviewer()
+        assert "build" in str(reviewer.state_dir)
```
Comment (mvillmow) [inline tests/unit/automation/test_audit_reviewer.py:413]: Test name says it verifies `dry_run` is passed to `gh_pr_review_post`, but the assertion checks `mock_coord.call_args[1]["dry_run"] is True` — this only validates the coordinator call, not the posting step. Either rename to `test_dry_run_passes_dry_run_to_coordinator` or add an assertion that `mock_post` was called with `dry_run=True` (requires patching `gh_pr_review_post` in this test).

**Review by mvillmow — COMMENTED** [review]
POLICY VIOLATION: auto-merge enabled (must be disabled until GO); PR description empty (missing Closes #994); commit signing state array is empty (no signed commits recorded).

File: hephaestus/automation/github_api.py (hunk 1 above)
Comment (mvillmow) [reply hephaestus/automation/github_api.py]: Fixed: `fetch_open_prs()` already uses `--limit 2147483647` (INT_MAX) in `hephaestus/automation/github_api.py:1259-1260` on this branch — the fix was applied in a prior commit. Verified with `git show origin/994-auto-impl:hephaestus/automation/github_api.py | grep limit`. No code change needed in this commit.

File: hephaestus/automation/audit_reviewer.py (hunk 2 above)
Comment (mvillmow) [reply hephaestus/automation/audit_reviewer.py]: Fixed: The guard in `audit_reviewer.py:177-179` already uses the correct `if response.strip() and not audits: raise RuntimeError(...)` pattern — this was applied in a prior commit. The `{"not_audits": 42}` case (no `audits` key, no `pr_number` key) correctly raises now. Verified by reviewing the current source and the test at `test_nonempty_response_zero_parses_raises`.

File: hephaestus/automation/audit_reviewer.py (hunk 3 above)
Comment (mvillmow) [reply hephaestus/automation/audit_reviewer.py]: Fixed: `--pr-numbers` already uses `nargs="+"` in `audit_reviewer.py:238` — applied in a prior commit. Bare `--pr-numbers` without values now correctly errors with "expected at least one argument" rather than silently auditing all PRs.

File: hephaestus/automation/audit_reviewer.py (hunk 4 above)
Comment (mvillmow) [reply hephaestus/automation/audit_reviewer.py:136]: Verified: `issue=0` is safe. `session_name()` in `session_naming.py:153` does `str(issue).lstrip("#").strip()` — `str(0)` = `"0"`, which passes the non-empty guard. The same call pattern is used in `pr_reviewer.py:225` with actual issue numbers. `0` produces the session name `"{repo}_0_{agent}"`, which is valid and distinct from any real issue. The comment at `audit_reviewer.py:149-151` accurately documents this.

File: hephaestus/automation/audit_reviewer.py (hunk 5 above)
Comment (mvillmow) [reply hephaestus/automation/audit_reviewer.py:180]: Fixed: write_audit_report is already gated behind if not self.dry_run at audit_reviewer.py:213-215. During --dry-run no report file is written.

File: tests/unit/automation/test_audit_reviewer.py (hunk 6 above)
Comment (mvillmow) [reply tests/unit/automation/test_audit_reviewer.py:413]: Fixed in commit d3fc177: test_dry_run_passes_dry_run_to_gh_pr_review_post now patches gh_pr_review_post and asserts mock_post.assert_called_once() and mock_post.call_args[1]["dry_run"] is True, verifying the posting step also receives dry_run=True.

**Review by mvillmow — COMMENTED** [review]
Addressed all 6 correctness review threads:

1. **--limit 0** (PRRT_kwDOQww0as6HmJ7e): Already fixed on branch — `fetch_open_prs()` uses `--limit 2147483647`.
2. **Guard condition** (PRRT_kwDOQww0as6HmJ7f): Already fixed — `if response.strip() and not audits: raise` is the correct pattern.
3. **nargs="*"** (PRRT_kwDOQww0as6HmJ7g): Already fixed — uses `nargs="+"`.
4. **issue=0** (PRRT_kwDOQww0as6HmMF1): Verified safe — `session_name()` converts 0→"0" (non-empty); documented in source comment.
5. **write_audit_report dry_run** (PRRT_kwDOQww0as6HmMF3): Already fixed — gated behind `if not self.dry_run`.
6. **Test assertion** (PRRT_kwDOQww0as6HmN8n): Fixed in d3fc177 — now patches `gh_pr_review_post` and asserts `dry_run=True` on the posting step.

Also fixed pre-existing CI failures (mypy type errors, CLI table sync, omit allowlist count).
